### PR TITLE
feat: Believable asset tracker shadows

### PR DESCRIPTION
**Pull request by sherretten**
## Problem
The shadow data created by the simulator is very old, based on Asset Tracker v1.

## Solution
Update it to create typical data that would be written by the currently supported Asset Tracker v2, or the Multi-Service Sample. 

Added CLI option `-t` to specify ATv2(atv2) or MSS(mss). It will default to atv2 if incorrect/no input.

## Testing
**local**
```sh
# Run a device, or create a new device and specify mss or atv2
npm run build && node dist/cli.js -k <api key> -a -t <mss or atv2>
```
The resulting shadow differences will be:
ATv2 includes a `config` object while MSS won't.

Other differences
| value | MSS | ATv2 |
| ------ | ------ | ------ |
| device.deviceInfo.appName |  'nrf_cloud_multi_service' |  'asset_tracker_v2' |
| device.serviceInfo.fota_v2 | ['MODEM', 'APP'] | ['BOOT', 'MODEM', 'APP'] |

**Jira**
https://nordicsemi.atlassian.net/browse/IRIS-8767

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ These are the options. Most of them are set with environment variables.
   -o, --device-ownership-code <deviceOwnershipCode>  PIN/ownership code of the device (default: "123456")
   -m, --mqtt-messages-prefix <mqttMessagesPrefix>    The prefix for the MQTT for this tenant for sending and receiving device messages (default: "")
   -v, --verbose                                      Output debug information
+  -t, --asset-tracker <assetTracker>                 Type of asset tracker. "mss" or "atv2" (default: "atv2")
```
Comment (jayteemo) [inline README.md]:
```suggestion
  -t, --app-type <appType>                 Type of device application. "mss" or "atv2" (default: "atv2")                             
```

File: src/models/Device.ts [hunk 2]
```diff
@@ -139,59 +140,66 @@ export class NrfDevice {
...
+								batteryVoltage: 5191,
+								appName: assetTracker === 'mss' ? 'nrf_cloud_multi_service' : 'asset_tracker_v2',
+								imei: '358299840010349',
+								board: 'nrf9161dk_nrf9161',
+								sdkVer: 'v2.6.0-571-gf927cd6b1473',
+								zephyrVer: 'v3.5.99-ncs1-4957-g54b4e400ed8f',
+								hwVer: 'nRF9161 LACA ADA'
+							},
+							networkInfo: {
+								supportedBands: assetTracker === 'mss' ? '(1,2,3,4,5,8,12,13,18,19,20,25,26,28,66,85)' : '',
```
Comment (jayteemo) [inline src/models/Device.ts]: i dont think there is much of a reason to handle `supportedBands` and `networkMode` differently.
i would say just always use `'(1,2,3,4,5,8,12,13,18,19,20,25,26,28,66,85)'` and `LTE-M GPS`

**Review by jayteemo — COMMENTED** [review]
i believe

**Review by plskeggs — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -63,7 +63,7 @@ These are the options. Most of them are set with environment variables.
   -o, --device-ownership-code <deviceOwnershipCode>  PIN/ownership code of the device (default: "123456")
   -m, --mqtt-messages-prefix <mqttMessagesPrefix>    The prefix for the MQTT for this tenant for sending and receiving device messages (default: "")
   -v, --verbose                                      Output debug information
-  -t, --asset-tracker <assetTracker>                 Type of asset tracker. "mss" or "atv2" (default: "atv2")                             
+  -t, --app-type <appType>                 					 Type of app type. "mss" or "atv2" (default: "atv2")
```
Comment (jayteemo) [inline README.md]: type of app type?
🤷

**Review by jayteemo — APPROVED** [review]

File: README.md
Comment (lincolnthomas) [inline README.md:1]: I just noticed, please change line 160 to what it actually says now:
```
connecting to "mqtt.nrfcloud.com"...
```

File: src/cli.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ const getConfig = (env: any, args: string[]): SimulatorConfig =>
     )
     .option(
       '-s, --services <services>',
-      'Comma-delimited list of services to enable. Any of: [gps,acc,temp,device]',
+      'Comma-delimited list of services to enable. Any of: [gps,acc,temp,device,rsrp,location,log,alert]',
```
Comment (lincolnthomas) [inline src/cli.ts:37]: good catch

File: src/models/Device.ts [hunk 5]
```diff
@@ -139,59 +140,66 @@ export class NrfDevice {
...
-								'GPS',
-								'FLIP',
-								'TEMP',
-								'HUMID',
-								'AIR_PRESS',
-								'BUTTON',
-								'LIGHT',
-								'LOG',
-							],
+	//Change this to handle atv2 and mss.
```
Comment (lincolnthomas) [inline src/models/Device.ts]: Sounds like a TODO, which it's not. 
```suggestion
	// Create a shadow for either Asset Tracker v2 (atv2) or Multi-Service Sample (mss)
```

File: src/models/Device.ts [hunk 6]
```diff
@@ -139,59 +140,66 @@ export class NrfDevice {
...
+								movementTimeout: 3600,
+								accThreshAct: 4,
+								accThreshInact: 4,
+								accTimeoutInact: 60,
+								nod: []
+							}
+						}),
+						device: {
+							deviceInfo: {
+								appVersion: appVersion ?? "1.0.0",
```
Comment (lincolnthomas) [inline src/models/Device.ts:172]: good

**Review by lincolnthomas — CHANGES_REQUESTED** [review]
Code looks good to me, minor nits, but running it, there's no `config` object for atv2 (mss correctly doesn't have it). For example:

```
node dist/cli.js -k ___ -d mydev-atv2-2 -v -h https://api.dev.nrfcloud.com -a -s temp
```
created the shadow on Dev:
```
{
  "state": {
    "desired": {
      "pairing": {
        "state": "paired",
        "topics": {
          "d2c": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/m/d/mydev-atv2-2/d2c",
          "c2d": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/m/d/mydev-atv2-2/+/r"
        }
      },
      "nrfcloud_mqtt_topic_prefix": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/"
    },
    "reported": {
      "connection": {
        "status": "disconnected",
        "keepalive": 30,
        "disconnectReason": "CONNECTION_LOST",
        "clientInitiatedDisconnect": false
      },
      "control": {
        "alertsEn": true,
        "logLvl": 3
      },
      "device": {
        "deviceInfo": {
          "appVersion": "1",
          "batteryVoltage": 5191,
          "appName": "asset_tracker_v2",
          "imei": "358299840010349",
          "board": "nrf9161dk_nrf9161",
          "sdkVer": "v2.6.0-571-gf927cd6b1473",
          "zephyrVer": "v3.5.99-ncs1-4957-g54b4e400ed8f",
          "hwVer": "nRF9161 LACA ADA"
        },
        "networkInfo": {
          "supportedBands": "(1,2,3,4,5,8,12,13,18,19,20,25,26,28,66,85)",
          "networkMode": "LTE-M GPS",
          "ipAddress": "10.160.33.51",
          "ueMode": 2,
          "rsrp": -58
        },
        "simInfo": {
          "uiccMode": 1,
          "iccid": "",
          "imsi": "204080813516718"
        },
        "serviceInfo": {
          "fota_v2": [
            "BOOT",
            "MODEM",
            "APP"
          ]
        },
        "connectionInfo": {
          "protocol": "MQTT",
          "method": "LTE"
        }
      }
    },
    "delta": {
      "pairing": {
        "state": "paired",
        "topics": {
          "d2c": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/m/d/mydev-atv2-2/d2c",
          "c2d": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/m/d/mydev-atv2-2/+/r"
        }
      },
      "nrfcloud_mqtt_topic_prefix": "dev/8da963d3-f0f0-432d-a667-20bfdc0f3145/"
    }
  }
}
```

**Review by lincolnthomas — APPROVED** [review]
I ran it again with the latest from this branch, and the atv2 version now has the expected `config` object. 👍 
For the record from @sherretten : 

> when I changed the name from assetTracker -> appName, I didn't update the name throughout, so the shadow creator was getting undefined  and I had removed the default value from the method.
Fix: change all occurances to appName, and add back in the default value of atv2.

**Pull request merged**